### PR TITLE
[Workflows] Top and left sections of new Client Profile

### PR DESCRIPTION
**Pull request by macfarlandian**
## Description of the change

Adds a new client profile page. The page is excluded from navigation for now so we can iterate without blocking deployments, but you can access it by editing the URL of any existing form-fill page (just change "compliantReporting" to "client"). Also builds out about two-thirds of the initial UI for the page; I didn't want to further bloat this PR so I will follow up with the remainder separately.

There is some refactoring here for code-sharing and reorganization but the existing pages should remain basically unchanged (other than splitting up the Contact and Housing sections in the profile).

## Type of change

- [x] New feature (non-breaking change that adds functionality)

## Related issues

> Closes https://github.com/Recidiviz/recidiviz-data/issues/12637, closes https://github.com/Recidiviz/recidiviz-data/issues/12799, closes https://github.com/Recidiviz/recidiviz-data/issues/12638

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- ~E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))~

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Review by Catacola — APPROVED** [review]
I only have minor comments which can be addressed in future PRs (and may already be).

File: src/core/PracticesClientProfile/SupervisionProgress.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+  const lastDay = startOfDay(expirationDate);
+
+  const monthDiff = Math.abs(differenceInMonths(lastDay, today));
+
+  const durationFromExp = intervalToDuration({ start: today, end: lastDay });
+  return `${formatDuration(durationFromExp, {
+    format: monthDiff < 6 ? ["months", "days"] : ["years", "months"],
+    delimiter: ", ",
+  })} ${today > lastDay ? "past EXP" : "to go"}`;
+}
```
[the remark this answers is not in the document]
Comment (macfarlandian) [reply src/core/PracticesClientProfile/SupervisionProgress.tsx:99]: yes that was intentional!

File: src/core/WorkflowsLayouts/WorkflowsFormLayout.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  &:first-child {
+    border-bottom: 1px solid ${palette.slate20};
+  }
+`;
+
+const SidebarWrapper: React.FC = ({ children }) => {
+  return (
+    <Sidebar>
+      <SidebarSection>
+        <RecidivizLogo />
```
[the remark this answers is not in the document]
Comment (macfarlandian) [reply src/core/WorkflowsLayouts/WorkflowsFormLayout.tsx:58]: yeah good call, that makes sense to me

[the remark this answers is not in the document]
Comment (macfarlandian) [reply src/core/WorkflowsLayouts/WorkflowsFormLayout.tsx:58]: I will address in a future PR though once more of the URL routing is in place

**Pull request merged**